chia/consensus/blockchain.py

```diff
@@ -262,16 +262,19 @@ async def _reconsider_peak(
 
                 # Begins a transaction, because we want to ensure that the coin store and block store are only updated
                 # in sync.
-                await self.block_store.begin_transaction()
+                await self.block_store.db_wrapper.lock.acquire()
```

Comment:
Is the finally clause reached in case of an exception?

I much rather do:
`self.block_store.db_wrapper.lock`
So we don't need a finally clause and we know that lock is released

Reply:
yup, finally gets called always. (even with re-raise in except:)